content/en/platform/corda/5.0-dev-preview-2/developing/api/api-persistence.md

```diff
@@ -8,3 +8,173 @@ menu:
+
+Liquibase manages DB changes in a “Change Log” which references one or more Change Sets.
+You must specify the top level `databaseChangeLog` in a resource file in the CPK called `migration/db.changelog-master.xml`.
+This file can reference one or more files including `changeSet`.
+
+You should organise these change sets with future changes in mind.
+For example, we recommend a single `include` per version of the table.
+Once a `changeSet` is deployed, it cannot be changed and any change must be provided as a `changeSet` with a new `id`.
+We suggest adding a version in the id, for example, `<table-name>-v1`.
+
+Example of `src/resources/migration/db.changelog-master.xml`:
+```xml
+<?xml version="1.1" encoding="UTF-8" standalone="no"?>
+<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+                   xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                   xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-4.3.xsd">
+    <include file="migration/dogs-migration-v1.0.xml"/>
+</databaseChangeLog>
+```
+The referenced `include` file should also be a resource file in `src/resources/migration` and define the table itself. For example, `src/resources/migration/dogs-migration-v1.0.xml`:
+```xml
+<?xml version="1.1" encoding="UTF-8" standalone="no"?>
+<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+                   xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                   xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-4.3.xsd">
+
+    <changeSet author="R3.Corda" id="dogs-migrations-v1.0">
+        <createTable tableName="dog">
+            <column name="id" type="uuid">
+                <constraints nullable="false"/>
+            </column>
+            <column name="name" type="VARCHAR(255)"/>
+            <column name="birthdate" type="DATETIME"/>
+            <column name="owner" type="VARCHAR(255)"/>
+        </createTable>
+        <addPrimaryKey columnNames="id" constraintName="dog_id" tableName="dog"/>
+    </changeSet>
+</databaseChangeLog>
+```
+{{< note >}}
+The include file reference is resolved relative to the resources path in the CPK and not relative to the current directory.
+{{< /note >}}
+
+## Running the migrations
+
+To run the migrations:
+1. Upload the CPI.
+2. Create a virtual node using the [Corda 5 REST RPC interface](../../developing/rest-api/rest-api.html).
+
+  The migrations run when the virtual node is created and logging shows the migrations executing.
+  If you have direct database access, you should see the tables being created.
+  If you are using Postgres, make sure to look under the correct schema, since each virtual node creates a new schema, unless an external VNode DB was provided during VNode creation.
```

Comment:
```suggestion
  If you are using Postgres, make sure to look under the correct schema, since each virtual node creates a new schema, unless an external VNode database was provided during VNode creation.
```